src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final String COLOR = new ColorSupplier().getRandomColor();
+    private static final int LIMIT_NUMBER = 20;
+
+    public Figure getRandomFigure() {
+        switch (RANDOM.nextInt(4)) {

```

Comment:
The Random number generation for the switch case should be in the range of 0 to 4 because you have 5 cases (0 to 4). Use `RANDOM.nextInt(5)` instead of `RANDOM.nextInt(4)`.